src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -108,20 +108,20 @@ public static ParseResult Parse(string expressionText, ParserOptions options = n
         public CheckResult Check(string expressionText, RecordType parameterType = null, ParserOptions options = null)
         {
             var parse = Parse(expressionText, options);
-
-            var bindingConfig = new BindingConfig(options?.AllowsSideEffects == true);
-            return CheckInternal(parse, parameterType, bindingConfig);
+            return Check(parse, parameterType, options);
         }
 
         /// <summary>
         /// Type check a formula without executing it. 
         /// </summary>
         /// <param name="parse">the parsed expression. Obtain from <see cref="Parse(string, ParserOptions)"/>.</param>
         /// <param name="parameterType">types of additional args to pass.</param>
+        /// <param name="options">parser options to use.</param>
         /// <returns></returns>
-        public CheckResult Check(ParseResult parse, RecordType parameterType = null)
+        public CheckResult Check(ParseResult parse, RecordType parameterType = null, ParserOptions options = null)
```

Comment:
Since it's probably weird to have `ParserOptions` on a method that does not perform any parsing, an alternative would be to store them together with `ParseResult`. Thoughts?